zk/model/config.py

```diff
@@ -0,0 +1,37 @@
+""" Configuration key store, elements such as event_city or contact_email """
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.dialects.postgresql import JSON # Required - unsure why
+from meta import Base, Session, metadata
+
+import logging
+log = logging.getLogger(__name__)
+
+class Config(Base):
+    __tablename__ = 'config'
+    category = sa.Column(sa.types.String, primary_key=True)
+    key      = sa.Column(sa.types.String, primary_key=True)
+    value    = sa.Column(sa.dialects.postgresql.JSON)
+    # NOTE: Non-postgresql requires the use of a decorator
+
+    @classmethod
+    def get(cls, category, key=None):
```

Comment:
Hi David,

I would expect sql alchemy to make caching reasonably straight forward, but I've not looked at that recently.

I'll see what I can find when I look at the migration.

James

> On 8 May 2015, at 08:28, David Tulloh <notifications@github.com> wrote:
> 
> In zk/model/config.py:
> 
> > +from sqlalchemy import orm
> > +from sqlalchemy.dialects.postgresql import JSON # Required - unsure why
> > +from meta import Base, Session, metadata
> > +
> > +import logging
> > +log = logging.getLogger(__name__)
> > +
> > +class Config(Base):
> > +    __tablename__ = 'config'
> > +    category = sa.Column(sa.types.String, primary_key=True)
> > +    key      = sa.Column(sa.types.String, primary_key=True)
> > +    value    = sa.Column(sa.dialects.postgresql.JSON)
> > +    # NOTE: Non-postgresql requires the use of a decorator
> > +
> > +    @classmethod
> > +    def get(cls, category, key=None):
> I would prefer to add caching as another commit.
> 
> I think it will be a solid chunk of work and have a bunch of issues unrelated to this merge.
> 
> —
> Reply to this email directly or view it on GitHub.
>